presto-raptor/src/test/java/com/facebook/presto/raptor/storage/organization/TestShardOrganizationManager.java

```diff
@@ -0,0 +1,197 @@
+    public void teardown()
+    {
+        dummyHandle.close();
+    }
+
+    @Test
+    public void testOrganizationEligibleTables()
+            throws Exception
+    {
+        long table1 = metadataDao.insertTable("schema", "table1", false, true, null);
+        metadataDao.insertColumn(table1, 1, "foo", 1, "bigint", 1, null);
+
+        metadataDao.insertTable("schema", "table2", false, true, null);
+        metadataDao.insertTable("schema", "table3", false, false, null);
+        assertEquals(metadataDao.getOrganizationEligibleTables(), ImmutableSet.of(table1));
+    }
+
+    @Test(timeOut = 1_000)
+    public void testTableDiscovery()
+            throws Exception
+    {
+        long table1 = metadataDao.insertTable("schema", "table1", false, true, null);
+        metadataDao.insertColumn(table1, 1, "foo", 1, "bigint", 1, null);
+
+        long table2 = metadataDao.insertTable("schema", "table2", false, true, null);
+        metadataDao.insertColumn(table2, 1, "foo", 1, "bigint", 1, null);
+
+        metadataDao.insertTable("schema", "table3", false, false, null);
+
+        ShardOrganizationManager organizationManager = createShardOrganizationManager("100ms");
+
+        // initializes tables
+        Set<Long> actual = organizationManager.discoverAndInitializeTablesToOrganize();
+        assertEquals(actual, ImmutableSet.of(table1, table2));
+
+        // update the start times and test that the tables are discovered after interval seconds
+        organizerDao.updateLastStartTime("node1", table1);
+        organizerDao.updateLastStartTime("node1", table2);
+
+        // wait for some time (interval time) for the tables to be eligible for organization
+        while (organizationManager.discoverAndInitializeTablesToOrganize().isEmpty()) {
+            MILLISECONDS.sleep(10);
+        }
+        assertEquals(organizationManager.discoverAndInitializeTablesToOrganize(), ImmutableSet.of(table1, table2));
+    }
+
+    @Test
+    public void testSimple()
+            throws Exception
+    {
+        long timestamp = Duration.ofDays(Duration.ofNanos(System.nanoTime()).toDays()).toMillis();
+        int day = Ints.checkedCast(Duration.ofNanos(System.nanoTime()).toDays());
+
+        List<ShardIndexInfo> shards = ImmutableList.of(
+                shardWithRange(1, ShardRange.of(new Tuple(types, 5L, "hello", day, timestamp), new Tuple(types, 10L, "hello", day, timestamp))),
+                shardWithRange(1, ShardRange.of(new Tuple(types, 7L, "hello", day, timestamp), new Tuple(types, 10L, "hello", day, timestamp))),
+                shardWithRange(1, ShardRange.of(new Tuple(types, 6L, "hello", day, timestamp), new Tuple(types, 9L, "hello", day, timestamp))),
+                shardWithRange(1, ShardRange.of(new Tuple(types, 1L, "hello", day, timestamp), new Tuple(types, 5L, "hello", day, timestamp))));
+
+        Set<OrganizationSet> actual = createOrganizationSets(tableInfo, shards);
+
+        assertEquals(actual.size(), 1);
```

Comment:
Add a comment explaining what this is asserting. I don't understand the `0, 1, 2`